2015/proposals/banaszkiewicz-piotr.md

```diff
@@ -0,0 +1,158 @@
+for lower footprint and easier maintainance (and because I feel comfortable
+using them).
+
+API, in a RESTful fashion, will be implemented using
+[django-tastypie](http://tastypieapi.org/).
+
+Rough database structure:
+
+* system information table
+    * operating system family (Windows / MacOSX / Linux / Other)
+    * OS version (8.1 / 10.9 / Ubuntu / ?)
+    * Hardware architecture (x86_64 / x86 / Other)
+* package checks table
+    * package name
+    * requested package version
+    * found package version
+    * failed check?
+    * fail reason
+
+Every entry in both tables will be additionally assigned a
+[universally unique identifier](http://en.wikipedia.org/wiki/Universally_unique_identifier).
+Every single UUID will correspond to one diagnostic data upload.
+
+Additionally we might want to store workshop-specific data, ie.:
+
+* track admissions from specific workshops
+* track packages requested for specific workshop.
+
+Enhancements to the installation testing script will use standard tools
+available in Python Standard Library and on students' systems (like, for
+example, `uname`).
+
+I want to use Mozilla's
+[Metrics-Graphics](https://github.com/mozilla/metrics-graphics) for charts and
+graphs, because this JavaScript graphing seems actively developed.
+
+## Schedule of Deliverables
+
+A word on my availability:
+
+In the month of June I have to prepare for exams. I'll make sure to be
+spending at least 20hrs per week on this project.
+
+My exams are not scheduled yet, but the timeslot for them is
+**June 23rd - July 7th**. In this time I won't be able to work on the project.
+However, most likely I'll be over with the exams by the end of June - in that
+case I'll resume working on the project as soon as I pass everything.
+
+First I want to start with installation testing script. I want to make it
+gather diagnostic data from the system. This way if my proposed database
+structure is wrong at some point, I can easily change it. (This might be
+thought of as a test-driven development.)
+
+### May 25th - June 7th
+
+Implement gathering of operating system diagnostic data. Start testing that
+script on at least one MacOSX machine, couple Windows boxes, and as many Linux
+boxes as possible.
+
+### June 8th - June 21st
+
+Continue testing. Change database schema if required. Implement REST API.
+Start implementing tests for that API.
+
+### June 22nd - July 5th
+
+I'm unavailable due to exams.
+
+### July 6th - July 19th
+
+Provided that REST API is stable and has good (100%) test coverage, start
+working on a front end for Software Carpentry admins. Most likely graphs,
+charts, and so on will take the biggest amount of work at this point.
+
+### July 20th - August 2nd
+
+Finish up UI, probably have a round of UX testing with Software Carpentry admins.
+
+### August 3rd - August 16th
+
+Implement sending diagnostic data in installation testing script. Write
+documentation.
```

Comment:
I don't expect this will take long.  Personally, I'd be tempted to handle this part after stubbing in the API and before adding API tests, so you could run the script to populate the database (and then use those entries for test fixtures, etc.).  Your call though.